crates/ascot-os/src/mac.rs

```diff
@@ -0,0 +1,327 @@
+#[cfg(target_os = "linux")]
+mod linux {
+    use std::fs;
+    use std::path::Path;
+
+    use tracing::warn;
+
+    const IFACE_TYPE_ETHERNET: u16 = 1;
+    const IFACE_TYPE_WIFI: u16 = 801;
+
+    // Known MAC OUIs for common virtual machine vendors.
+    // Source: https://standards-oui.ieee.org/oui.txt and common known VM vendors.
+    const VM_MAC_PREFIXES: &[[u8; 3]] = &[
+        [0x00, 0x05, 0x69], // VMware
+        [0x00, 0x0C, 0x29], // VMware
+        [0x00, 0x1C, 0x14], // VMware
+        [0x00, 0x50, 0x56], // VMware
+        [0x00, 0x03, 0xFF], // Microsoft Hyper-V
+        [0x00, 0x15, 0x5D], // Microsoft Hyper-V
+        [0x08, 0x00, 0x27], // Oracle VirtualBox
+        [0x0A, 0x00, 0x27], // Oracle VirtualBox
+        [0x00, 0x1C, 0x42], // Parallels
+    ];
+
+    // Checks if the MAC address is locally administered (bit 1 of the first byte is set).
+    // Locally administered addresses are usually assigned by software and not globally unique hardware addresses.
+    fn is_locally_administered_mac(mac: [u8; 6]) -> bool {
+        (mac[0] & 0x02) != 0
+    }
+
+    // Checks if the MAC address is from a known virtual machine vendor based on MAC OUI prefix.
+    // Returns true if the MAC prefix matches known virtual adapters (VMware, Hyper-V, VirtualBox, etc.).
+    fn is_virtual_mac_vendor(mac: [u8; 6]) -> bool {
+        VM_MAC_PREFIXES.iter().any(|prefix| prefix == &mac[0..3])
+    }
+
+    fn is_virtual_interface(iface_path: &Path, mac: [u8; 6]) -> bool {
+        // If the interface does not have a "device" entry, it is considered virtual.
+        if !iface_path.join("device").exists() {
+            return true;
+        }
+
+        if is_locally_administered_mac(mac) {
+            return true;
+        }
+
+        if is_virtual_mac_vendor(mac) {
+            return true;
+        }
+
+        // Canonical path checks for virtual devices or hypervisor interfaces.
+        if let Ok(canon) = fs::canonicalize(iface_path) {
+            if let Some(s) = canon.to_str() {
+                // Virtual devices path.
+                if s.contains("/sys/devices/virtual/") {
+                    return true;
+                }
+
+                // Hyper-V virtual interface.
+                if s.contains("VMBUS") {
+                    return true;
+                }
+
+                // Other hypervisors/virtualizations? Add strings here if needed:
+                // For example, KVM devices often under /sys/devices/virtual/net or specific patterns,
+                // but usually covered by "/sys/devices/virtual/" check.
```

Comment:
We can change this comment explaining `how` and `why` is necessary to add  hypervisors/virtualizations here. Something like:

`Other  hypervisors/virtualizations can be added here. The strings representing their values can be evaluated ...
`

Just a model on how to write the comment